File: CLI/card.py
cardNo = int(input("Enter card number : "))
cardStr = str(cardNo)

slicedArray = []
evenArray = []
oddArray = []

def arraySeprator(subArray:list, digits:str):
    for i in digits:
        doubledDigit = int(i)*2
        if len(str(doubledDigit)) > 1:
            for j in str(doubledDigit):
                subArray.append(int(j))
        else:
            subArray.append(int(doubledDigit))

evenDigits = cardStr[-2::-2][::-1]
oddDigits = cardStr[-1::-2][::-1]

for i in str(oddDigits):
    oddArray.append(int(i))

arraySeprator(evenArray, evenDigits)

evenSum = sum(evenArray)
oddSum = sum(oddArray)

totalSum = evenSum + oddSum

if totalSum % 10 == 0:
    print("Card is valid")
else:
    print("Card is not valid")

File: CLI/card.cpp
#include <iostream>
#include <string>

int evenSum(const std::string cardNum) {

    int sum = 0;

    for (int i = -2; i >= 0; i -= 2) {
        int doubleNum = (cardNum[i] - '0') * 2;
        if (doubleNum > 9) {
            for (int j = 0; j < std::to_string(doubleNum).length(); j++) {
                sum += (std::to_string(doubleNum)[i] - '0');
            }
        } else {
            sum += doubleNum;
        }
    }

    return sum;
}

int oddSum(const std::string cardNum) {
    int sum = 0;
    
    for (int i = -1; i >= 0; i -= 2) {
        sum += cardNum[i] - '0';
    }

    return sum;
}

int main() {

    long long cardNum;

    std::cout << "Enter card number : ";
    std::cin >> cardNum;

    std::string cardStr = std::to_string(cardNum);

    int even = evenSum(cardStr);
    int odd = oddSum(cardStr);

    int total = even + odd;

    if (total % 10 == 0) {
        std::cout << "Card is Valid";
    } else {
        std::cout << "Card is not Valid";
    }

    return 0;
}